addons/website/static/src/snippets/s_share/000.js

```diff
@@ -77,7 +76,15 @@ const ShareWidget = publicWidget.Widget.extend({
                 return a + title + c;
             })
             .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
+                // Some pages (/profile/user/ID) don't have an image to share.
+                const ogImageEl = document.querySelector("meta[property='og:image']");
+                if (ogImageEl) {
+                    const media = encodeURIComponent(ogImageEl.content);
+                    return a + media + c;
+                }
+                const params = new URLSearchParams(a + c);
+                params.delete("media");
+                return `?${params.toString()}`;
```

Comment:
I don't mind the use of `URLSearchParams` but could you check it? I tried with `"https://www.hello.com/world?foo=bar&media={media}&yes=no"`, this seems to return `https://www.hello.com/world?foo=baryes=no`...

Reply:
Indeed, it's fixed @qsm-odoo